### PR TITLE
Redesign: grid view

### LINKED ISSUE #6206: Redesign: Gridview
## Feature Description

Update page grid view mode based on [Figma](https://www.figma.com/file/bMhG3KyrJF8vIAODgmbeqT/Design-System?node-id=298%3A1113).

- [x] Update background opacity (should be fully opaque)
- [x] Update back button (using design system button)
- [x] Update slider (blocked by #4938)
- [x] Update spacing and sizing for all elements in view
- [x] Update to new focus rectangle (could be generalized from carousel page focus rectangle)
- [x] Update drag and drop styling (it might need native drag and drop rather than current fake, but that would go in a separate less important ticket)

| Current design | New design |
|-|-|
| Screen Shot 2021-02-04 at 22 48 25 | Screen Shot 2021-02-18 at 9 15 46 PM |

---

_Do not alter or remove anything below. The following sections will be managed by moderators only._

### PR DESCRIPTION
**Pull request by barklund**
This implements the re-designed grid view mode.

## To-do

See todo in #6206.

## User-facing changes

| Before | After |
|-|-|
| Screen Shot 2021-02-24 at 18 10 31 | Screen Shot 2021-02-24 at 18 10 01 |

### QA

This PR can be tested by following these steps:

1. Add multiple pages to a story (and make sure they're distinguishable)
2. Open the grid view
3. Interact and observe it working as per spec

### UAT

- [x] UAT should use the same steps as above.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testiing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #6206

### CODE REVIEW

**Comment by miina** [comment]
Works well!
Some notes from testing:
- Not sure if the design changed meanwhile but in Fimga, it looks like the background color is `bg.black` and
- the slider also seems to have a different color background in the grid view.
- Slider has a tooltip in Figma:
Screenshot 2021-02-26 at 09 55 11

One more thing noticed (not on Figma though) is that the margin 122px between the slider and the pages feels large when using a small screen, perhaps something to flag with UX and leave for later improvement if needed?
Screenshot 2021-02-26 at 09 58 45

**Comment by miina** [comment]
@googlebot I consent.

**Comment by barklund** [comment]
> Works well!

I've forwarded your UX questions to the channel. And I've fixed the tooltip (it was actually there, just behind the modal) by putting tooltip z-index to the top.

File: assets/src/edit-story/components/colorPicker/editablePreview.js [hunk 1]
```diff
@@ -32,7 +32,6 @@ import {
 } from '../../../design-system';
 
 const Preview = styled.button`
-  padding: 0;
```
Comment (miina) [inline assets/src/edit-story/components/colorPicker/editablePreview.js]: Hm, how is this related 🤔, from some merge? It should have the `0`, otherwise the preview will look different from the edit mode.

Comment (barklund) [reply assets/src/edit-story/components/colorPicker/editablePreview.js]: Yes, it was from a (bad) merge. 0 is the right one?

**Review by miina — APPROVED** [review]
LGTM (pending the color picker seemingly unrelated change)

File: assets/src/edit-story/components/colorPicker/editablePreview.js (hunk 1 above)
Comment (miina) [reply assets/src/edit-story/components/colorPicker/editablePreview.js]: Yes, 0 is the right one.
